cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -160,6 +160,9 @@ func Run(s *options.CMServer) error {
 	if err != nil {
 		return err
 	}
+	// it is more safe to use hostname plus pod name
+	// see https://github.com/kubernetes/kubernetes/issues/22775
+	id = id + "-" + os.Getenv("POD_NAME")
```

Comment:
Using container port would prevent they are scheduled on the same node, right?

Using hostname(podname) is enough for disambiguation. ^_^